release/models/interfaces/openconfig-if-common.yang

```diff
@@ -0,0 +1,301 @@
+        including framing characters.
+
+        Discontinuities in the value of this counter can occur
+        at re-initialization of the management system, and at
+        other times as indicated by the value of
+        'last-clear'.";
+      reference
+        "RFC 2863: The Interfaces Group MIB - ifHCInOctets.
+        RFC 4293: Management Information Base for the
+        Internet Protocol (IP).";
+    }
+
+    leaf in-pkts {
+      type oc-yang:counter64;
+      description
+        "The total number of packets received on the interface,
+        including all unicast, multicast, broadcast and bad packets
+        etc.";
+      reference
+        "RFC 2819: Remote Network Monitoring Management Information Base.
+        RFC 4293: Management Information Base for the
+        Internet Protocol (IP).";
+    }
+
+    leaf in-unicast-pkts {
+      type oc-yang:counter64;
+      description
+        "The number of packets, delivered by this sub-layer to a
+        higher (sub-)layer, that were not addressed to a
+        multicast or broadcast address at this sub-layer.
+
+        Discontinuities in the value of this counter can occur
+        at re-initialization of the management system, and at
+        other times as indicated by the value of
+        'last-clear'.";
+      reference
+        "RFC 2863: The Interfaces Group MIB - ifHCInUcastPkts.
+        RFC 4293: Management Information Base for the
+        Internet Protocol (IP).";
+    }
+
+    leaf in-broadcast-pkts {
+      type oc-yang:counter64;
+      description
+        "The number of packets, delivered by this sub-layer to a
+        higher (sub-)layer, that were addressed to a broadcast
+        address at this sub-layer.
+
+        Discontinuities in the value of this counter can occur
+        at re-initialization of the management system, and at
+        other times as indicated by the value of
+        'last-clear'.";
+      reference
+        "RFC 2863: The Interfaces Group MIB - ifHCInBroadcastPkts.
+        RFC 4293: Management Information Base for the
+        Internet Protocol (IP).";
+    }
+
+    leaf in-multicast-pkts {
+      type oc-yang:counter64;
+      description
+        "The number of packets, delivered by this sub-layer to a
+        higher (sub-)layer, that were addressed to a multicast
+        address at this sub-layer.  For a MAC-layer protocol,
+        this includes both Group and Functional addresses.
+
+        Discontinuities in the value of this counter can occur
+        at re-initialization of the management system, and at
+        other times as indicated by the value of
+        'last-clear'.";
+      reference
+        "RFC 2863: The Interfaces Group MIB - ifHCInMulticastPkts.
+        RFC 4293: Management Information Base for the
+        Internet Protocol (IP).";
+    }
+
+    leaf in-errors {
+      type oc-yang:counter64;
+      // TODO: this counter combines several error conditions --
```

Comment:
Is this `TODO` still relevant?

Reply:
I'd like to complain a little bit that this is expanding the scope of the PR which is intended to be small and only seeks to
- add multicast counters
- remove physical interface errors counters from subinterfaces which are invalid

Thank you for reading my complaint.  :)

That said, I investigated and I think this TODO can be removed.  Here are the error counters which seem to match commonly supported output from 'show interfaces' commands on various implementations.  There is variation by This varies by hardware and NOS.  Some platforms report only input and output errors without support for additional, more granular errors.  Some platforms aggregate errors as input and output errors and also support more granular errors.  

* [Survey of JunOS platforms and different outputs](https://www.juniper.net/documentation/us/en/software/junos/routing-policy/topics/ref/command/show-interfaces-statistics.html#show-interfaces-statistics-command__d84034e266)
* [Cisco 8000 series interface errors](https://www.cisco.com/c/en/us/td/docs/iosxr/cisco8000/Interfaces/b-interfaces-hardware-component-cr-8000/m-global-interface-commands.html)
* 

Common error counters | OC Path | Description
-- | -- | --
runts | [...ethernet/state/counters/in-undersize-frames](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-ethernet-state-counters-in-undersize-frames) | Number of received packets that were too small to be handled. Aggregated to input errors count.
giants | [.../ethernet/state/counters/in-oversize-frames](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-ethernet-state-counters-in-oversize-frames) | Number of received packets that were too large. Aggregated to input errors count.
ignored |  [.../state/counters/in-discards](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-in-discards) | Total number of ignored packet errors. Ignored packets are those that are discarded because the interface hardware does not have enough internal buffers. Broadcast storms and bursts of noise can result in an increased number of ignored packets.
throttles |  [.../state/counters/in-discards](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-in-discards) | Number of packets dropped due to throttling (because the input queue was full).
input errors | [.../state/counters/in-errors](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-in-errors) | Total number of received packets that contain errors and hence cannot be delivered. Compare this to total input drops, which counts packets that were not delivered despite containing no errors.
CRC |  [/interfaces/interface/ethernet/state/counters/in-crc-errors](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-ethernet-state-counters-in-crc-errors)<br />  | Number of packets that failed the Ethernet CRC check.
frame | [/interfaces/interface/state/counters/in-fcs-errors](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-in-fcs-errors)|  Number of packets with bad framing bytes.  Perhaps further investigation needed to de-duplicate the in-crc-errors and in-fcs-errors leaves.
overrun | [/interfaces/interface/state/counters/out-discards](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-out-discards)| Number of overrun errors experienced by the interface. Overruns represent the number of times that the receiver hardware is unable to send received data to a hardware buffer because the input rate exceeds the receiver's ability to handle the data.
resets | Number of times that the hardware has been reset. The triggers and effects of this event are hardware-specifc.
output errors |  [/interfaces/interface/state/counters/out-discards](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-out-discards) | Number of times that the receiver hardware was unable to handle received data to a hardware buffer because the input rate exceeded the receiver's ability to handle the data.
underruns | [/interfaces/interface/state/counters/out-discards](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-out-discards) |  Underruns will occur when the low level transceiver tries to send data, but the buffer is still empty because the device hasn't serviced the packets yet.
output buffer failures | [/interfaces/interface/state/counters/out-discards](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-out-discards) | Number of times that a packet was not output from the output hold queue because of a shortage of MEMD shared memory.
carrier transitions | [interfaces/interface/state/counters/carrier-transitions](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-state-counters-carrier-transitions) | Number of times the interface state has transitioned between up and down.

Reply:
addressed in other comment

Reply:
For future reference -- you can just say "no, but we can address that in a future change" :-) Since we generally only look at these TODOs either in a fixit or just when we change the code, I'll often ask about them as we change that line.

Reply:
That being said - thanks for looking into it!